lib/noether/either.ex

```diff
@@ -17,10 +17,51 @@ defmodule Noether.Either do
     iex> map({:error, "Value not found"}, &Kernel.abs/1)
     {:error, "Value not found"}
   """
-  @spec map(either(), fun()) :: either()
-  def map({:ok, a}, f), do: {:ok, f.(a)}
+  @spec map(either(), fun1()) :: either()
+  def map({:ok, a}, f) when is_function(f, 1), do: {:ok, f.(a)}
   def map(any = {:error, _}, _), do: any
 
+  @doc """
+  Given an `{:ok, {:ok, value}}` it flat the ok unwrapping the `value` and returning `{:ok, value}`.
```

Comment:
I think the example has a typo: `it flattens the ok unwrapping...` rather than `it flat the ok unwrapping`